editor/src/com/talosvfx/talos/editor/addons/scene/assets/AssetRepository.java

```diff
@@ -84,14 +84,14 @@ public class AssetRepository extends BaseAssetRepository implements Observer {
 	public static final AssetNameFieldFilter ASSET_NAME_FIELD_FILTER = new AssetNameFieldFilter();
 
 	private ObjectMap<GameAssetType, ObjectMap<String, GameAsset<?>>> identifierGameAssetMap = new ObjectMap<>();
+	private ObjectMap<GameAssetType, ObjectMap<String, GameAsset<?>>> uniqueIdentifierGameAssetMap = new ObjectMap<>();
```

Comment:
Why not use UUID object itself? Because of this, you have to call toString() of UUID objects in the runtime. 
As equals() function is overridden in the UUID class you shouldn't have problems I guess.

Reply:
Fixed.